cocos/core/data/garbage-collection.ts

```diff
@@ -27,14 +27,33 @@ declare class FinalizationRegistry {
     unregister (token: any);
 }
 
+const targetSymbol = Symbol('[[target]]');
+
 class GarbageCollectionManager {
     private _finalizationRegistry: FinalizationRegistry | null = EDITOR ? new FinalizationRegistry(this.finalizationRegistryCallback.bind(this)) : null;
+    private _gcObjects: WeakMap<any, GCObject> = new WeakMap();
 
     public registerGCObject (gcObject: GCObject): GCObject {
         if (EDITOR) {
-            gcObject._finalizationToken = {};
-            const proxy = new Proxy(gcObject, {});
-            this._finalizationRegistry!.register(proxy, gcObject, gcObject._finalizationToken);
+            const token = {};
+            const proxy = new Proxy(gcObject, {
+                get (target, property, receiver) {
+                    if (property === targetSymbol) {
+                        return target;
+                    }
+                    let val = Reflect.get(target, property, receiver);
+                    if (typeof val === 'function' && property !== 'constructor') {
+                        const original = val;
+                        val = function () {
+                            return original.apply((this as GCObject)[targetSymbol], arguments);
```

Comment:
Deconstruct the wrapper of `this`,  Avoid circular references to proxy objects when internal method invoked